unittests/utilities/test_fec_timer.py

```diff
@@ -58,6 +66,33 @@ def test_error(self):
                     found = True
             assert found
 
+    def test_nested(self):
+        FecTimer.start_category(FecTimer.WAITING)
+        FecTimer.start_category(FecTimer.RUN_OTHER)
+        FecTimer.start_category(FecTimer.MAPPING)
+        FecTimer.start_category(FecTimer.GET_MACHINE, True)
+        FecTimer.end_category(FecTimer.GET_MACHINE)
+        FecTimer.end_category(FecTimer.MAPPING)
+        FecTimer.end_category(FecTimer.RUN_OTHER)
+        on, off = ProvenanceReader().get_category_timer_sums(
+            FecTimer.RUN_OTHER)
+        total = ProvenanceReader().get_category_timer_sum(
+            FecTimer.RUN_OTHER)
+        self.assertGreater(on, 0)
+        self.assertGreater(off, 0)
+        self.assertEqual(total, on + off)
+        on, off = ProvenanceReader().get_category_timer_sums(
+            FecTimer.MAPPING)
+        self.assertGreater(on, 0)
+        self.assertGreater(off, 0)
+
+    def test_mess(self):
+        FecTimer.end_category(FecTimer.WAITING)
+        FecTimer.start_category(FecTimer.RUN_OTHER)
+        FecTimer.start_category(FecTimer.MAPPING)
+        FecTimer.end_category(FecTimer.RUN_OTHER)
+        FecTimer.end_category(FecTimer.MAPPING)
```

Comment:
Shouldn't there be an error here?

Reply:
it does Error but does it with just a log and changing to ERROR.

Discussed with @rowleya and decided to actually Error!